runtime/v2/app.go

```diff
@@ -126,3 +126,11 @@ func (a *App) GetLogger() log.Logger {
 func (a *App) ExecuteGenesisTx(_ []byte) error {
 	panic("not implemented")
 }
+
+func (a *App) SetAppVersion(context.Context, uint64) error {
```

Comment:
Can't we leave those unimplemented on the app?
The version modifier can be nil, so the logic will be skipped anyway.

Reply:
We could flag this optional and remove:
https://github.com/cosmos/cosmos-sdk/blob/2a2b3f15dc1bd3871498d9bd16e7b9c6c91ce5a1/x/upgrade/depinject.go#L42
If you're OK with that approach

Reply:
I don't understand why keeping them returning zero values then 🤔